### PR TITLE
Sensor Expiration Timer

### PR DESCRIPTION
**Pull request by IsThisPaul**
These updates add a Sensor Expiration Timer and Date to the CGM Settings page.

### CODE REVIEW

**Review by ps2 — CHANGES_REQUESTED** [review]
Looks good. I'm not a fan of the label-less second row, but it should be ok for now; the real fix would probably be to update the UI to the SwiftUI based design style that's being used on other devices, but that's a bigger piece of work.

Only change request is to make `sessionExpTime` a computed var, so readers of the code see it's computed, and not an independent variable.

**Comment by IsThisPaul** [comment]
I reverted the commit where I moved sessionExpTime and it is now a clearly calculated variable (in the view controller) again.

**Comment by ps2** [comment]
@IsThisPaul The expiration time makes sense to put on the Glucose struct, especially since you'll have one place it's defined, instead of copied twice as you have it in the VC.  I just meant it might read a little better if it was a computed var.  I.e.:
```
    var sessionExpDate: {
        return sessionStartTime + (10*24*60*60)
    }
```

**Comment by IsThisPaul** [comment]
I don't think I totally follow what you're saying. I might need some hand holding on it.

File: CGMBLEKitUI/TransmitterSettingsViewController.swift [hunk 1]
```diff
@@ -158,10 +160,40 @@ class TransmitterSettingsViewController: UITableViewController {
         formatter.doesRelativeDateFormatting = true
         return formatter
     }()
-
+    
+    private lazy var sensorExpFullFormatter: DateFormatter = {
+        let formatter = DateFormatter()
+        formatter.dateFormat = "E, MMM d 'at' h:mm a"
```
Comment (ps2) [inline CGMBLEKitUI/TransmitterSettingsViewController.swift]: This is non-localized.  Maybe just use `dateFormatter` above.

File: CGMBLEKitUI/TransmitterSettingsViewController.swift [hunk 2]
```diff
@@ -115,7 +115,9 @@ class TransmitterSettingsViewController: UITableViewController {
     }
 
     private enum AgeRow: Int, CaseIterable {
-        case sensor
+        case sensorage
```
Comment (ps2) [inline CGMBLEKitUI/TransmitterSettingsViewController.swift]: camel case and full words: `sensorAge`, `sensorCountdown`, `sensorExpirationDate`

File: CGMBLEKitUI/TransmitterSettingsViewController.swift [hunk 3]
```diff
@@ -158,10 +160,40 @@ class TransmitterSettingsViewController: UITableViewController {
...
     }()
-
+    
+    private lazy var sensorExpFullFormatter: DateFormatter = {
+        let formatter = DateFormatter()
+        formatter.dateFormat = "E, MMM d 'at' h:mm a"
+        return formatter
+    }()
+    
+    private lazy var sensorExpRelFormatter: DateFormatter = {
```
Comment (ps2) [inline CGMBLEKitUI/TransmitterSettingsViewController.swift]: Avoid abbreviations in variable names.

File: CGMBLEKitUI/TransmitterSettingsViewController.swift [hunk 4]
```diff
@@ -158,10 +160,40 @@ class TransmitterSettingsViewController: UITableViewController {
...
     private lazy var sessionLengthFormatter: DateComponentsFormatter = {
         let formatter = DateComponentsFormatter()
-        formatter.allowedUnits = [.day, .hour]
+        formatter.allowedUnits = [.day, .hour, .minute]
+        formatter.unitsStyle = .full
+        formatter.maximumUnitCount = 2
+        return formatter
+    }()
+    
+    private lazy var sessionLengthMinsFormatter: DateComponentsFormatter = {
```
Comment (ps2) [inline CGMBLEKitUI/TransmitterSettingsViewController.swift]: This is not used

File: CGMBLEKitUI/TransmitterSettingsViewController.swift (hunk 1 above)
Comment (IsThisPaul) [reply CGMBLEKitUI/TransmitterSettingsViewController.swift]: I use this formatter to display the day of the week, which dateFormatter does not. I've modified this formatter to use ".full" dateStyle and ".short" timeStyle. If we could find a way to localize this though, A briefer, abbreviated date might look a little cleaner.

File: CGMBLEKitUI/TransmitterSettingsViewController.swift (hunk 2 above)
Comment (IsThisPaul) [reply CGMBLEKitUI/TransmitterSettingsViewController.swift]: Done.

File: CGMBLEKitUI/TransmitterSettingsViewController.swift (hunk 3 above)
Comment (IsThisPaul) [reply CGMBLEKitUI/TransmitterSettingsViewController.swift]: Fixed.

File: CGMBLEKitUI/TransmitterSettingsViewController.swift (hunk 4 above)
Comment (IsThisPaul) [reply CGMBLEKitUI/TransmitterSettingsViewController.swift]: Removed.

**Comment by IsThisPaul** [comment]
> @IsThisPaul The expiration time makes sense to put on the Glucose struct, especially since you'll have one place it's defined, instead of copied twice as you have it in the VC. I just meant it might read a little better if it was a computed var. I.e.:
> 
> ```
>     var sessionExpDate: {
>         return sessionStartTime + (10*24*60*60)
>     }
> ```

Cleaned this, and the other issues up per our conversation.

**Pull request merged**